### PR TITLE
feat(medusa): Expose an `activeManager_` getter in TransactionBaseService

### PR DESCRIPTION
**Pull request by adrien2p**

### CODE REVIEW

File: packages/medusa/src/services/sales-channel-inventory.ts [hunk 1]
```diff
@@ -40,14 +40,13 @@ class SalesChannelInventoryService {
...
   ): Promise<number> {
-    const locations = await this.salesChannelLocationService_.listLocations(
-      salesChannelId
-    )
-
-    return await this.inventoryService_.retrieveAvailableQuantity(
-      inventoryItemId,
-      locations
-    )
+    const locations = await this.salesChannelLocationService_
```
Comment (adrien2p) [inline packages/medusa/src/services/sales-channel-inventory.ts:43]: self-review: @carlos-r-l-rodrigues here the transaction was not passed along for those calls, plus this service was not supporting the transactions. I did some changes, can you let me know what you think

File: packages/medusa/src/services/product-variant-inventory.ts [hunk 2]
```diff
@@ -636,6 +615,10 @@ class ProductVariantInventoryService extends TransactionBaseService {
         // first get all inventory items required for a variant
         const variantInventory = await this.listByVariant(variant.id)
 
+        const salesChannelInventoryServiceTx =
```
Comment (adrien2p) [inline packages/medusa/src/services/product-variant-inventory.ts:618]: self-review: same here @carlos-r-l-rodrigues I did the same changes

File: packages/medusa/src/services/cart.ts [hunk 3]
```diff
@@ -2193,20 +2190,19 @@ class CartService extends TransactionBaseService {
...
   ): Promise<void> {
-    const transactionManager = this.transactionManager_ ?? this.manager_
-
     // If the cart contains items, we update the price of the items
     // to match the updated region or customer id (keeping the old
     // value if it exists)
     if (cart.items?.length) {
       const region = await this.regionService_
-        .withTransaction(this.transactionManager_)
+        .withTransaction()
```
Comment (carlos-r-l-rodrigues) [inline packages/medusa/src/services/cart.ts:2198]:
```suggestion
        .withTransaction(this.activeManager_)
```

File: packages/medusa/src/services/sales-channel-inventory.ts (hunk 1 above)
Comment (carlos-r-l-rodrigues) [reply packages/medusa/src/services/sales-channel-inventory.ts:43]: looks fine. thanks

**Review by carlos-r-l-rodrigues — APPROVED** [review]
LGTM. Nice cleanup

**Review by olivermrbl — APPROVED** [review]
Such a nice clean up! @srindom can I get you to sign off on this too :)

**Comment by olivermrbl** [comment]
@adrien2p Should be update to use new Typeorm repository API

**Comment by adrien2p** [comment]
> @adrien2p Should be updated to use the new Typeorm repository API

@olivermrbl as soon as develop will be merged into it and I resolve the conflict it will use the new repo 💪

**Comment by olivermrbl** [comment]
> as soon as develop will be merged into it 

Yep sorry, that was what I meant 😅

**Review by srindom — COMMENTED** [review]
Super nice! The only thing that I can think of as a potential risk with this is if people have done:
```ts
// my-custom-service.ts

import { TransactionBaseService } from "@medusajs/medusa"

class MyService extends TransactionBaseService {
 ...
}
```

TS would complain after an update right? So this is a breaking change that should be highlighted.

**Comment by adrien2p** [comment]
> Super nice! The only thing that I can think of as a potential risk with this is if people have done:
> 
> ```ts
> // my-custom-service.ts
> 
> import { TransactionBaseService } from "@medusajs/medusa"
> 
> class MyService extends TransactionBaseService {
>  ...
> }
> ```
> 
> TS would complain after an update right? So this is a breaking change that should be highlighted.

I am not sure what would TS complain about?

**Comment by srindom** [comment]
Was unsure if the change from `protected abstract` -> `protected` would make the compiler complain, but if it doesn't then all good 💪

**Comment by adrien2p** [comment]
> Was unsure if the change from `protected abstract` -> `protected` would make the compiler complain, but if it doesn't then all good 💪

@srindom it does not complain because now it is just optional to define it on the child class, but if it is present it is not a problem since protected member are accessible from the child

**Pull request merged**